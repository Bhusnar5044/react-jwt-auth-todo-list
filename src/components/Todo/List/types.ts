export type Task = { text: string; done: boolean };

export type Props = {
    tasks: Task[];
    setTasks: React.Dispatch<React.SetStateAction<Task[]>>;
};
