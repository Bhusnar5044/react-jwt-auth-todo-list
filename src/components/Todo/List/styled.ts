import { styled } from '@utils';

export const StyledEmpty = styled.div`
    margin-top: 2rem;
    p {
        margin: 0;
        margin-bottom: 1rem;
    }
    svg {
        display: inline-block;
        width: 50px;
        opacity: 0.5;
    }
`;

export const StyledList = styled.ul`
    display: flex;
    flex-direction: column;
    gap: 1rem;
    list-style: none;
    margin: 0;
    padding: 0;
`;
