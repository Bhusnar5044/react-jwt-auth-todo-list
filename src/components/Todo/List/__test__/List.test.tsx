import { render } from '@test-utils';
import { List } from '../List';

describe('List', () => {
    it('should render properly', () => {
        const { container } = render(<List tasks={[]} setTasks={jest.fn()} />);
        expect(container).toMatchSnapshot();
    });
});
