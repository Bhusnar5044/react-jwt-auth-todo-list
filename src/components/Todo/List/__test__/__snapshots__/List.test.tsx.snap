// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should render properly 1`] = `
.c0 {
  margin-top: 2rem;
}

.c0 p {
  margin: 0;
  margin-bottom: 1rem;
}

.c0 svg {
  display: inline-block;
  width: 50px;
  opacity: 0.5;
}

<div>
  <div
    class="c0 empty"
  >
    <p>
      <svg
        viewBox="0 0 21 21"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          fill-rule="evenodd"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          transform="translate(2.5 4.5)"
        >
          <path
            d="m3.65939616 0h8.68120764c.4000282 0 .7615663.23839685.9191451.6060807l2.7402511 6.3939193v4c0 1.1045695-.8954305 2-2 2h-12c-1.1045695 0-2-.8954305-2-2v-4l2.74025113-6.3939193c.15757879-.36768385.51911692-.6060807.91914503-.6060807z"
          />
          <path
            d="m0 7h4c.55228475 0 1 .44771525 1 1v1c0 .55228475.44771525 1 1 1h4c.5522847 0 1-.44771525 1-1v-1c0-.55228475.4477153-1 1-1h4"
          />
        </g>
      </svg>
    </p>
    <p>
      <b>
        Empty list
      </b>
    </p>
    <p>
      Add a new task above
    </p>
  </div>
</div>
`;
